File: smart_update/src/main/java/com/tang/smartupdate/PatchUtil.java
package com.tang.smartupdate;

/**
 * Created by tanghongtu on 2020/6/23.
 */
public class PatchUtil {

//    static {
//        System.loadLibrary("SmartUpdate");
//    }

    /**
     * @param oldApkPath
     * @param newApkPath
     * @param patchPath
     * @return 0 成功,  1 失败
     */
    public static native int patch(String oldApkPath, String newApkPath, String patchPath);

}

File: smart-update/src/main/java/com/tang/smartupdate/PatchUtil.kt
package com.tang.smartupdate

/**
 * Created by tanghongtu on 2020/6/23.
 */
object PatchUtil {

    init {
        System.loadLibrary("smart-update")
    }
    /**
     * @param oldApkPath
     * @param newApkPath
     * @param patchPath
     * @return 0 成功,  1 失败
     */
    external fun patch(oldApkPath: String, newApkPath: String, patchPath: String): Int
}